File: LeetCode/single-number/solutions/main-naive.go
func singleNumber(nums []int) int {
    var counter map[int]int = make(map[int]int, len(nums))

	for _, num := range nums {
		counter[num]++
	}

	for k, v := range counter {
		if v == 1 {
			return k
		}
	}

	return 0
}

File: LeetCode/single-number/solutions/main-naive.js
const SingleNumber = (nums) => {
  const numsObj = {};

  for (let i = 0; i < nums.length; i++) {
    numsObj[nums[i].toString()] = (numsObj[nums[i].toString()] || 0) + 1;
  }

  for (let key in numsObj) {
    if (numsObj[key] === 1) return +key;
  }
};
